app/Http/Controllers/Api/RoleController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Api\RolePermissionRequest;
+use App\Http\Requests\Api\RoleRequest;
+use Spatie\Permission\Models\Role;
+
+class RoleController extends Controller
+{
+    public function index()
+    {
+        $roles = Role::all();
+
+        return response($roles);
+    }
+
+    public function store(RoleRequest $request)
+    {
+        $validated = $request->validated();
+        $role = Role::create($validated);
+
+        return response($role);
+    }
+
+    public function show(Role $role)
+    {
+        return $role;
```

Comment:
what's the difference between `response($role)` and `$role`?

Reply:
So the difference between `response($role)` and `$role`  is that `response($role)` allows to create a custom response by wrapping the `$role` data and the `$role`, laravel will be automatically convert to the HTTP response with a JSON response.